src/Entity.php

```diff
@@ -585,4 +585,23 @@ protected function getResolvedFields(): array
     {
         return static::fields();
     }
+
+    /**
+     * Make new entity instance with copied fields
+     *
+     * @return Entity
```

Comment:
You can change this to `static`, that gives a better indication of what is getting returned. In PHP8 it is even supported to use `static` as a return type for the method, but that's for another day. If you change it to `@return static` you don't need the annotation anymore in the test for the copy.